packages/terra-date-time-picker/src/DateTimeUtils.js

```diff
@@ -48,10 +48,15 @@ class DateTimeUtils {
 
   static syncDateTime(momentDate, iOSdate, time) {
     const date = moment(iOSdate);
+
+    if (!date.isValid()) {
+      return momentDate;
+    }
+
     let newDate = momentDate ? momentDate.clone() : date;
 
     // If momentDate was null, a new moment date needs to be created and sync'd with the entered time.
-    if (momentDate === null && time && time.length === 5) {
+    if (time && time.length === 5) {
```

Comment:
Should we update the inline comment since we no longer check if `momentDate === null`

Reply:
Definitely. https://github.com/cerner/terra-framework/pull/619/commits/3512b63fa2bb09f6ee9e79b42000edc697dc7237

Reply:
I'm a little confused, but its likely my lack of code knowledge for this module. 

We are updating to return the moment date if the generated iSOdate is invalid but shouldn't we still attempt to update the time regardless of if the iOSdate or momentDate's value if we are updating the time when date is invalid?

Reply:
I agree that _generally_ if the iSOdate is invalid, it should proceed to update the (base) moment object using the time. In our use case, if the iSOdate is invalid (the date input is cleared), the moment object in our state doesn't really care about the time. I have updated it to account for the general use case to proceed to update the time even if the date is invalid https://github.com/cerner/terra-framework/pull/619/commits/784e52a725189be07a1cc7da27ca567b630385ed